data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumerTest.java

```diff
@@ -0,0 +1,139 @@
+
+    @Mock
+    private KafkaConsumer<String, Object> avroConsumer;
+    private AtomicBoolean status;
+
+    @Mock
+    private Buffer<Record<Object>> buffer;
+
+    @Mock
+    private KafkaSourceConfig sourceConfig;
+
+    private TopicConfig topicConfig;
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    private String schemaType;
+    private KafkaSourceCustomConsumer consumer;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        Yaml yaml = new Yaml();
+        FileReader fileReader = new FileReader(getClass().getClassLoader().getResource("sample-pipelines.yaml").getFile());
+        Object data = yaml.load(fileReader);
+        if (data instanceof Map) {
+            Map<String, Object> propertyMap = (Map<String, Object>) data;
+            Map<String, Object> logPipelineMap = (Map<String, Object>) propertyMap.get("log-pipeline");
+            Map<String, Object> sourceMap = (Map<String, Object>) logPipelineMap.get("source");
+            Map<String, Object> kafkaConfigMap = (Map<String, Object>) sourceMap.get("kafka");
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.registerModule(new JavaTimeModule());
+            String json = mapper.writeValueAsString(kafkaConfigMap);
+            Reader reader = new StringReader(json);
+            sourceConfig = mapper.readValue(reader, KafkaSourceConfig.class);
+            topicConfig = sourceConfig.getTopics().get(0);
+        }
+        pluginMetrics = mock(PluginMetrics.class);
+        buffer = mock(Buffer.class);
+        status = new AtomicBoolean();
+        plainTextConsumer = mock(KafkaConsumer.class);
+        jsonConsumer = mock(KafkaConsumer.class);
+        avroConsumer = mock(KafkaConsumer.class);
+        consumer = new KafkaSourceCustomConsumer(plainTextConsumer, status, buffer, topicConfig, sourceConfig, schemaType, pluginMetrics);
+    }
+
+    @Test
+    public void testConsumeRecords() throws InterruptedException {
+        consumer = new KafkaSourceCustomConsumer(plainTextConsumer, status, buffer, topicConfig, sourceConfig, schemaType, pluginMetrics);
+        String topic = topicConfig.getName();
+        schemaType = "plaintext";
+        Map<TopicPartition, List<ConsumerRecord<String, Object>>> records = new LinkedHashMap<>();
+        Thread producerThread = new Thread(() -> {
+            ConsumerRecord<String, Object> record1 = new ConsumerRecord<>(topic, 0, 0L, "mykey-1", "myvalue-1");
+            ConsumerRecord<String, Object> record2 = new ConsumerRecord<>(topic, 0, 0L, "mykey-2", "myvalue-2");
+            records.put(new TopicPartition(topic, 1), Arrays.asList(record1, record2));
+        });
+        producerThread.start();
+        TimeUnit.SECONDS.sleep(1);
+        producerThread.join();
+        KafkaSourceCustomConsumer spyConsumer = spy(consumer);
+        doCallRealMethod().when(spyConsumer).consumeRecords();
+        spyConsumer.consumeRecords();
+        verify(spyConsumer).consumeRecords();
```

Comment:
The tests should verify the consumed records. Just verifying a function called in line 104 that it is called in 105, is not a good test. Please verify that records are consumed as expected by some other means.

Reply:
we have added the suggested changes